pkg/apis/pipeline/v1alpha1/stepaction_validation.go

```diff
@@ -72,6 +73,18 @@ func (ss *StepActionSpec) Validate(ctx context.Context) (errs *apis.FieldError)
 	return errs
 }
 
+// validateNoParamSubstitutionsInScript validates that param substitutions are not invoked in the script
+func validateNoParamSubstitutionsInScript(script string) *apis.FieldError {
+	_, present, errString := substitution.ExtractVariablesFromString(script, "params")
+	if errString != "" || present {
```

Comment:
is it possible that present is false and errString is not nil?

Reply:
Not according to the implementation of the `substitution.ExtractVariablesFromString` function. There, either the match was not found (either if the regular expression failed to compile or there was no match). If a match was found, there could be an error in how the referencing was done (invalid param usage).